processing/src/main/java/org/apache/carbondata/processing/newflow/DataLoadProcessorStep.java

```diff
@@ -0,0 +1,40 @@
+package org.apache.carbondata.processing.newflow;
+
+import java.util.Iterator;
+
+import org.apache.carbondata.processing.newflow.exception.CarbonDataLoadingException;
+
+/**
+ * This base interface for data loading. It can do transformation jobs as per the implementation.
+ *
+ */
+public interface DataLoadProcessorStep {
```

Comment:
I think each implementation of this interface have similar logic in the execute function, can we create a abstract class to implement the common logic? 
The common logic like:
```
Iterator<Object[]> execute() throws CarbonDataLoadingException {
    Iterator childIter = child.execute();
    return new Iterator<Object[]> {
          public boolean hasNext() { 
                return childIter.hasNext();
          }
          public Object[] next() {
                // processInput is the abstract func in this class
                return processInput(childItor.next());
          }
     }
}
```